### PR TITLE
Closes #6373 #6374: Add logs to rucss

### LINKED ISSUE #6373: As a support team member, I want to know the last time each RUCSS-related CRON job started and ended so that I know if CRON are running as exepcted
**Context**

As part of building a RUCSS Debug tool, some information must be stored while in WP Rocket Debug mode so that it can be monitored easily by the support team.
More details here: https://www.notion.so/wpmedia/RUCSS-Issues-7adc4ed750fd4dc7902bb6881ed1204b?pvs=4#38398060400d4fcf97f27aa563483d68

**Expected behavior**
When in debug mode (WP_ROCKET_DEBUG?), a timestamp must be updated in the DB every time:
-  the CRON process_on_submit_jobs starts
-  the CRON process_on_submit_jobs ends successfully
- the CRON process_pending_jobs starts
- the CRON process_pending_jobs ends successfully
Those 4 timestamps must be stored in the DB and updated at every new related event.

When debug mode is off, nothing special happens. If the timestamps are in DB, they can stay there.

### LINKED ISSUE #6374: As a support team member, I want to know the last time a RUCSS-related Async action started and ended so that I know if AS is running as exepcted
**Context**

As part of building a RUCSS Debug tool, some information must be stored while in WP Rocket Debug mode so that it can be monitored easily by the support team.
More details here: https://www.notion.so/wpmedia/RUCSS-Issues-7adc4ed750fd4dc7902bb6881ed1204b?pvs=4#38398060400d4fcf97f27aa563483d68

**Expected behavior**

- When in debug mode (WP_ROCKET_DEBUG?), a timestamp must be updated in the DB every time the UsedCSS::check_job_status method is successful. We only need to keep the last time it happened.
When not in debug mode, nothing special happens. If there is a timestamp stored, it can stay there.

### PR DESCRIPTION
**Pull request by Miraeld**
## Description

The goal here is to add some logs in the DB for debugging purpose. 

Fixes #6373 #6374

## Type of change

- Enhancement (non-breaking change which improves an existing functionality).

## Is the solution different from the one proposed during the grooming?

No

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [ ] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

**Review by Khadreal — COMMENTED** [review]
Looks good to me

**Comment by mostafa-hisham** [comment]
Hi @Miraeld :vulcan_salute: from plugin team :grin:
I think we are missing call back saving the time of `rocket_rucss_process_on_submit_jobs_end` action in the [plugin PR](https://github.com/wp-media/wp-rocket/pull/6412/files)

and I don't know what are the options names for `RUCSS-related Async action started and ended` [issue-6374](https://github.com/wp-media/wp-rocket/issues/6374) I don't see them in the PR

We need those options to use them in the [RUCSS debug tool](https://github.com/orgs/wp-media/projects/112/views/2?pane=issue&itemId=53138212)

**Comment by Miraeld** [comment]
Hey @mostafa-hisham from service team :D

I've added the missing callback, and also added the check if debug is enabled as I forgot it. 

👍

**Comment by mostafa-hisham** [comment]
@Miraeld which options I should use to know
RUCSS-related Async action started
RUCSS-related Async action ended
to know if Action Scheduler is running as expected?

File: inc/Engine/Debug/RUCSS/Subscriber.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	 * @return void
+	 */
+	public function log_last_added_job_time( $is_success, $timestamp ) {
+		if ( Logger::debug_enabled() ) {
+			if ( ! $is_success ) {
+				return;
+			}
+
+			$this->options->set( 'last_rucss_job_added', $timestamp );
+			$this->options_api->set( 'debug', $this->options->get_options() );
```
Comment (CrochetFeve0251) [inline inc/Engine/Debug/RUCSS/Subscriber.php:66]: @Miraeld I am not sure what you wanted to save but here you are saving the whole wpr configurations inside the debug configuration

Comment (CrochetFeve0251) [reply inc/Engine/Debug/RUCSS/Subscriber.php:66]: This is due to the fact we are loading the wrong data into the options instance: https://github.com/wp-media/wp-rocket/blob/704f03428f286a9a84e32378626848444986534a/inc/classes/ServiceProvider/class-options.php#L36

**Review by CrochetFeve0251 — CHANGES_REQUESTED** [review]

File: inc/Engine/Debug/RUCSS/Subscriber.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+	/**
+	 * Saves the time when the process on submit jobs ended.
+	 *
+	 * @param string $timestamp Current timestamp.
+	 * @return void
+	 */
+	public function log_process_on_submit_end( $timestamp ) {
+		if ( Logger::debug_enabled() ) {
+			$this->options->set( 'log_process_on_submit_end', $timestamp );
```
Comment (mostafa-hisham) [inline inc/Engine/Debug/RUCSS/Subscriber.php]: @Miraeld 
Is this option name `log_process_on_submit_end` correct?

Comment (mostafa-hisham) [reply inc/Engine/Debug/RUCSS/Subscriber.php]: @Miraeld I think we can change it to `rucss_process_on_submit_jobs_end`

**Review by mostafa-hisham — CHANGES_REQUESTED** [review]

File: inc/Engine/Debug/RUCSS/Subscriber.php (hunk 2 above)
Comment (Miraeld) [reply inc/Engine/Debug/RUCSS/Subscriber.php]: It's been modified

**Pull request merged**